rasa/shared/core/flows/flow.py

```diff
@@ -373,13 +522,65 @@ def is_rasa_default_flow(self) -> bool:
         """Test whether something is a rasa default flow."""
         return self.id.startswith(RASA_DEFAULT_FLOW_PATTERN_PREFIX)
 
-    def get_collect_information_steps(self) -> List[CollectInformationFlowStep]:
+    def get_collect_steps(self) -> List[CollectInformationFlowStep]:
         """Return the collect information steps of the flow."""
-        collect_information_steps = []
+        collect_steps = []
         for step in self.steps:
             if isinstance(step, CollectInformationFlowStep):
-                collect_information_steps.append(step)
-        return collect_information_steps
+                collect_steps.append(step)
+        return collect_steps
+
+    @property
+    def steps(self) -> List[FlowStep]:
+        """Returns the steps of the flow."""
+        return self.step_sequence.steps
+
+
+@dataclass
+class StepSequence:
+    child_steps: List[FlowStep]
+
+    @staticmethod
+    def from_json(steps_config: List[Dict[Text, Any]]) -> StepSequence:
+        """Used to read steps from parsed YAML.
+
+        Args:
+            steps_config: The parsed YAML as a dictionary.
+
+        Returns:
+            The parsed steps.
+        """
+
+        flow_steps: List[FlowStep] = [step_from_json(config) for config in steps_config]
+
+        return StepSequence(child_steps=flow_steps)
+
+    def as_json(self) -> List[Dict[Text, Any]]:
+        """Returns the steps as a dictionary.
+
+        Returns:
+            The steps as a dictionary.
+        """
+        return [
+            step.as_json()
+            for step in self.child_steps
+            if not isinstance(step, InternalFlowStep)
+        ]
+
+    @property
+    def steps(self) -> List[FlowStep]:
```

Comment:
It's unclear to me the difference between `child_steps` and `steps` if both are Lists of `FlowSteps`, is the `steps` a flatter list?

Reply:
so `child_steps` are the immediate children of this step, while `steps` are all children, grandchildren, grandgrandchildren... So `steps` includes all steps that are a child of one of the child steps of this step and the step itself